tools/xnvme.c

```diff
@@ -623,6 +623,34 @@ sub_library_info(struct xnvmec *XNVME_UNUSED(cli))
 	return err;
 }
 
+static int
+sub_dsm(struct xnvmec *cli)
+{
+	struct xnvme_dev *dev = cli->args.dev;
+	struct xnvme_cmd_ctx ctx = xnvme_cmd_ctx_from_dev(dev);
+	uint32_t nsid = cli->args.nsid;
+	uint32_t nr = cli->args.nr;
+	uint32_t ad = cli->args.ad;
+	uint32_t idw = cli->args.idw;
+	uint32_t idr = cli->args.idr;
+	int err;
+
+	xnvmec_pinf("xNVMe DSM");
+	if (!cli->given[XNVMEC_OPT_NSID]) {
+		nsid = xnvme_dev_get_nsid(cli->args.dev);
+	}
+
```

Comment:
It should be possible to create at least one range when calling this.

Reply:
The CLI can now handle one range. The API can handle more